### PR TITLE
kodi: replace source urls for apache dependencies with stable ones

### PR DESCRIPTION
**Pull request by nvmd**
## Description of changes

Replaced source of apache `groovy`, `commons-lang3`, and `commons-text` with stable ones, they contain only the latest version:
> The most recent source and binary releases for the Apache Commons project are available from this directory listing. For older releases, please use the [archives](http://archive.apache.org/dist/commons/).

`commons-text` is already not there: `https://dlcdn.apache.org/commons/text/binaries/commons-text-1.11.0-bin.zip` returns 404, breaking the build of Kodi 20.1 just merged in #303497.

See in a build attempt of #304097: https://logs.ofborg.org/?key=nixos/nixpkgs.304097&attempt_id=45c6a77c-3113-46e8-a971-e5bd1f514559

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aanderse — APPROVED** [review]
😓

thank you

File: pkgs/applications/video/kodi/unwrapped.nix [hunk 1]
```diff
@@ -75,12 +75,12 @@ let
   };
 
   apache_commons_lang = fetchzip {
-    url = "https://dlcdn.apache.org//commons/lang/binaries/commons-lang3-3.14.0-bin.zip";
+    url = "https://archive.apache.org/dist/commons/lang/binaries/commons-lang3-3.14.0-bin.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/kodi/unwrapped.nix:78]:
```suggestion
    url = "mirror://apache/commons/lang/binaries/commons-lang3-3.14.0-bin.zip";
```

File: pkgs/applications/video/kodi/unwrapped.nix [hunk 2]
```diff
@@ -75,12 +75,12 @@ let
...
     sha512 = "sha512-eKF1IQ6PDtifb4pMHWQ2SYHIh0HbMi3qpc92lfbOo3uSsFJVR3n7JD0AdzrG17tLJQA4z5PGDhwyYw0rLeLsXw==";
   };
 
   apache_commons_text = fetchzip {
-    url = "https://dlcdn.apache.org//commons/text/binaries/commons-text-1.11.0-bin.zip";
+    url = "https://archive.apache.org/dist/commons/text/binaries/commons-text-1.11.0-bin.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/kodi/unwrapped.nix:83]:
```suggestion
    url = "mirror://apache/commons/text/binaries/commons-text-1.11.0-bin.zip";
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/video/kodi/unwrapped.nix (hunk 2 above)
Comment (nvmd) [reply pkgs/applications/video/kodi/unwrapped.nix]: Thanks! Didn't know such a thing existed. Is it documented somewhere?

What bugs me a little bit is that `fetchzip` succeeds only on the 5th attempt now. But that's the price to pay for "stability", I guess.

**Pull request merged**